scripts/prepare_disk.sh

```diff
@@ -22,6 +22,52 @@ usage()
     echo "NOTE: This script is destructive and will destroy your data."
 }
 
+is_resize_needed()
+{
+    is_comment()
+    {
+        test -z "${1%%#*}"
+    }
+
+    # Check if value is an integer, by using an inverse integer comparison test.
+    is_integer()
+    {
+        test "${1}" -eq "${1}" 2> /dev/null
+    }
+
+    current_partition_table_file="$(mktemp)"
+    sfdisk -d "${TARGET_DISK}" > "${current_partition_table_file}"
+
+    exit_loop=false
+    while IFS="${IFS}:=," read -r label _ start _ size _; do
+        if is_comment "${label}" || ! is_integer "${start}"; then
+            continue
+        fi
+
+        while IFS="${IFS}:=," read -r label2 _ start2 _ size2 _; do
+            if is_comment "${label2}" || ! is_integer "${start2}"; then
+                continue
+            fi
+
+            if [ "${label}" != "${label2}" ]; then
+                continue
+            fi
+
+            if [ "${start}" -ne "${start2}" ] || \
+               [ "${size}" -ne "${size2}" ]; then
+                exit_loop=true
+                return
```

Comment:
i assume break cannot be used as break will only exit one loop? and return will somehow exit both loops? Maybe add a short comment about this behavior? `# break both while loops`